File: vm/tests/all/interopSingleBasic/inheritMethods1.js
//JS

llvmFile = Polyglot.evalFile("llvm", "inheritMethods1.so");

a = new llvmFile.A(); //has methods foo1(), foo2()
b = new llvmFile.B(); //B extends A, has method foo2() only

console.log("a.foo1() should lead to: A::foo1");
a.foo1();
console.log("b.foo1() should lead to: A::foo1");
a.foo1();
console.log("a.foo2() should lead to: A::foo2");
a.foo2();
console.log("b.foo2() should lead to: B::foo2");
a.foo2();

File: vm/tests/all/interopSingleBasic/inheritMethods1.cpp
//CPP
#include<polyglot.h>
#include<iostream>
using namespace std;

class A {
	public:
		virtual void foo1();
		virtual void foo2();
};

POLYGLOT_DECLARE_CLASS(A);

A::foo1() {
	cout << "A::foo1()" << endl;
}
A::foo2() {
	cout << "A::foo2()" << endl;
}

class B:public A {
	public:
		void foo2();
};

POLYGLOT_DECLARE_CLASS(B);

B::foo2() {
	cout << "B::foo2()" << endl;
}
